src/pudl/analysis/record_linkage/eia_ferc1_record_linkage_splink_model.py

```diff
@@ -0,0 +1,388 @@
+        )
+        .sort_values(by="match_probability", ascending=False)
+        .groupby("record_id_ferc1")
+        .first()
+    )
+    preds_df = preds_df.reset_index()
+    train_df = inputs.get_train_df().reset_index()
+    true_pos = get_true_pos(preds_df, train_df)
+    false_pos = get_false_pos(preds_df, train_df)
+    false_neg = get_false_neg(preds_df, train_df)
+    logger.info(
+        "Metrics before overwrites:\n"
+        f"   True positives:  {true_pos}\n"
+        f"   False positives: {false_pos}\n"
+        f"   False negatives: {false_neg}\n"
+        f"   Precision:       {true_pos/(true_pos + false_pos):.03}\n"
+        f"   Recall:          {true_pos/(true_pos + false_neg):.03}\n"
+        "Precision = of the training data FERC records that the model predicted a match for, this percentage was correct.\n"
+        "A measure of accuracy when the model makes a prediction.\n"
+        "Recall = of all of the training data FERC records, the model predicted a match for this percentage.\n"
+        "A measure of coverage of FERC records."
+    )
+    experiment_tracker.execute_logging(
+        lambda: mlflow.log_metrics(
+            {
+                "precision": round(true_pos / (true_pos + false_pos), 3),
+                "recall": round(true_pos / (true_pos + false_neg), 3),
+            }
+        )
+    )
+    return preds_df
+
+
+@op(
+    out={
+        "out_pudl__yearly_assn_eia_ferc1_plant_parts_splink": Out(
+            # io_manager_key="pudl_io_manager"
+        )
+    }
+)
+def get_full_records_with_overwrites(best_match_df, inputs, experiment_tracker):
+    """Join full dataframe onto matches to make usable and get stats.
+
+    Overwrite the predictions dataframe with the training data, so that all
+    known bad predictions are corrected. Then join the EIA and FERC data on
+    so that the matches are usable. Drop model parameter and match probability
+    columns generated by splink. Log the coverge of the matches on the
+    FERC input data.
+    """
+    best_match_df = overwrite_bad_predictions(best_match_df, inputs.get_train_df())
+    connected_df = prettyify_best_matches(
+        matches_best=best_match_df,
+        plant_parts_eia_true=inputs.get_plant_parts_eia_true(),
+        plants_ferc1=inputs.get_plants_ferc1(),
+    )
+    _log_match_coverage(connected_df, experiment_tracker=experiment_tracker)
+    for match_set in ["all", "overrides"]:
+        check_match_consistency(
+            connected_df,
+            inputs.get_train_df(),
+            match_set=match_set,
+            experiment_tracker=experiment_tracker,
+        )
+    connected_df = add_null_overrides(
+        connected_df
+    )  # Override specified values with NA record_id_eia
+    return Resource.from_id(
+        "out_pudl__yearly_assn_eia_ferc1_plant_parts"
+    ).enforce_schema(connected_df)
+
+
+@models.pudl_model(
+    "out_pudl__yearly_assn_eia_ferc1_plant_parts_splink",
```

Comment:
This asset name will be updated once the existing PUDL model is replaced, it won't have the `splink` suffix.